src/server/chunk_server/run_chunk_server_main.cc

```diff
@@ -125,5 +122,8 @@ int main(int argc, char** argv) {
   // responsible for shutting down the server for this call to ever return.
   server->Wait();
 
+  // Terminate the reporting service
+  chunk_server_impl->TerminateReportToMaster();
```

Comment:
Just curious. Will this call actually be ever called. If we interrupt the thread to shut down this server, the thread may exit right away after cleanup, making this call never called.

Should we put this instead in the destructor of chunk server implementation instead?

Reply:
Yes I agree, lets have this in the destructor. It keeps things safe.

Reply:
Sounds good, moved to destructor.